src/arch/x86_64/CodeGen.zig

```diff
@@ -4252,3 +4343,7 @@ fn registerAlias(reg: Register, size_bytes: u32) Register {
         else => unreachable,
     }
 }
+
+fn abiSize(self: *Self, ty: Type) u64 {
+    return if (ty.isSlice()) 16 else ty.abiSize(self.target.*);
+}
```

Comment:
This should be fixed in type.zig, not hacked here in this x86 codegen file.

Reply:
Oh, this is a bug in type.zig? Oops, my bad! I thought this was by design. Lemme provide a fix!

Reply:
Done!